### ISSUE TITLE
Support Genrule in IjProjectSourcePathResolver breaks with Jargenrules

### ISSUE THREAD

**Issue by kageiit** [facebook/buck#2337]
This commit https://github.com/facebook/buck/commit/0673403d03c70d5cae3aad0348c25a70eced99b5#diff-74d224cad8bd2d6e8623905200b1d8e9 breaks when the abstractgenrule description is not of the the genruledescription type

For example:

```
java.lang.ClassCastException: com.facebook.buck.jvm.java.JarGenruleDescriptionArg cannot be cast to com.facebook.buck.shell.GenruleDescriptionArg
	at com.facebook.buck.features.project.intellij.IjProjectSourcePathResolver.getOutputPathForTargetNode(IjProjectSourcePathResolver.java:97)
```

cc @asp2insp

**Comment by asp2insp** [comment]
Thanks for reporting and apologies for the breakage! I'll put up a fix right away

**Comment by asp2insp** [comment]
Fixed in 8a21741a1b9d508dc3bf55a07d637413931e7e0d

**Comment by kageiit** [comment]
Thanks!
